packages/manager/apps/telecom/src/components/telecom/telephony/group/number/feature/ovhPabx/dialplan/dialplan.controller.js

```diff
@@ -2,55 +2,115 @@ import get from 'lodash/get';
 import set from 'lodash/set';
 import some from 'lodash/some';
 
-export default /* @ngInject */ function telephonyNumberOvhPabxDialplanCtrl(
-  $q,
-  $timeout,
-  $translate,
-  TucToast,
-  TUC_UI_SORTABLE_HELPERS,
-) {
-  const self = this;
-
-  self.loading = {
-    init: false,
-  };
-
-  self.popoverStatus = {
-    isOpen: false,
-    move: false,
-    rightPage: null,
-  };
-
-  self.displayHelpers = {
-    collapsed: false,
-    expanded: true,
-  };
-
-  self.sortableOptions = null;
-  self.ovhPabx = null;
-
-  /*= ==============================
-    =            HELPERS            =
-    =============================== */
-
-  self.isLoading = function isLoading() {
+export default class DialplanCtrl {
+  /* @ngInject */
+  constructor($q, $timeout, $translate, TucToast, TUC_UI_SORTABLE_HELPERS) {
+    this.$q = $q;
+    this.$timeout = $timeout;
+    this.$translate = $translate;
+    this.TucToast = TucToast;
+    this.tucUiSortableHelpers = TUC_UI_SORTABLE_HELPERS;
+  }
+
+  $onInit() {
+    this.loading = {
+      init: false,
+    };
+
+    this.popoverStatus = {
+      isOpen: false,
+      move: false,
+      rightPage: null,
+    };
+
+    this.displayHelpers = {
+      collapsed: false,
+      expanded: true,
+    };
+
+    this.sortableOptions = null;
+    this.ovhPabx = null;
+
+    let sortInterval = null;
+    let initPromise = this.$q.when(true);
+    this.loading.init = true;
+
+    // set ovhPabx instance
+    this.ovhPabx = this.numberCtrl.number.feature;
+
+    // set sortable options
+    this.sortableOptions = {
+      axis: 'y',
+      handle: '.extension-grip',
+      cancel: '.voip-plan__step-icon--grip-disabled',
+      containment: 'parent',
+      sort: this.tucUiSortableHelpers.variableHeightTolerance,
+      start() {
+        sortInterval = setInterval(() => {}, 33);
+      },
+      stop() {
+        if (sortInterval) {
+          clearInterval(sortInterval);
+        }
+      },
+      update() {
+        this.$timeout(() => {
+          // update extensions positions
+          angular.forEach(this.dialplan.extensions, (extension, index) => {
+            set(extension, 'position', index + 1);
+          });
+
+          // call api to update all positions
+          this.dialplan.updateExtensionsPositions();
+        });
+      },
+    };
+
+    if (this.dialplan) {
+      // load extension if dialplan exists
+      initPromise = this.dialplan.getExtensions();
+    } else if (this.numberCtrl.number.feature.dialplans.length === 0) {
+      initPromise = this.createDialplan();
+    }
```

Comment:
**suggestion**: Seems it could be done with a ternary operator

Reply:
Not for the moment